### PR TITLE
Rename to Extensions

### PR DESCRIPTION
**Pull request by mloufra**
Signed-off-by: mloufra <mloufra@amazon.com>

### Description
Change the class (PluginRequest, PluginResponse) to (InitializeExtensionsRequest, InitializeExtensionsResponse), and the variable name related to class (PluginRequest, PluginResponse).
Equivalent PR on OpenSearch : https://github.com/opensearch-project/OpenSearch/pull/4063

### Issues Resolved
https://github.com/opensearch-project/opensearch-sdk/issues/28

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/java/org/opensearch/sdk/ExtensionsRunner.java
Comment (dbwiddis) [inline src/main/java/org/opensearch/sdk/ExtensionsRunner.java:101]: The javadocs and log messages should be renamed consistently.

**Review by dbwiddis — DISMISSED** [review]
I think changing half of the instances of plugin to extension but leaving plugin around elsewhere is confusing.  Can we change the method names as well?

File: src/test/java/org/opensearch/sdk/TestExtensionsRunner.java
Comment (saratvemulapalli) [inline src/test/java/org/opensearch/sdk/TestExtensionsRunner.java:104]: Same as Dan suggested. 
Rename everything Plugin to Extension. 
```suggestion
        InitializeExtensionsResponse response = extensionsRunner.handleExtensionInitRequest(extensionInitRequest);
```

File: src/main/java/org/opensearch/sdk/ExtensionsRunner.java
Comment (owaiskazi19) [inline src/main/java/org/opensearch/sdk/ExtensionsRunner.java:109]: `new InitializeExtensionsResponse("RealExtension")` -> `new InitializeExtensionsResponse(extensionSettings.getExtensionName())`

**Review by owaiskazi19 — APPROVED** [review]

**Review by saratvemulapalli — APPROVED** [review]
Thanks @mloufra for the changes!!

**Comment by saratvemulapalli** [comment]
Looks like @dbwiddis requested changes, so we cannot merge this in without Dan's approval.

**Comment by owaiskazi19** [comment]
As build will fail for the other PRs. Dismissing @dbwiddis changes.

**Pull request merged**
